mvlearn/preprocessing/repeat.py

```diff
@@ -0,0 +1,201 @@
+        Either a single sklearn transformer that will be applied to each
+        view. One clone of the estimator will correspond to each view.
+        Otherwise, it should be a list of estimators, of length the number of
+        views in the multiview dataset.
+
+    Attributes
+    ----------
+    n_views_ : int
+        The number of views in the input dataset
+
+    transformer_list_ : list of objects of length n_views_
+        The list of transformer used to transform data. If
+        self.transformer is a single transformer, it is a list containing
+        clones of that transformer, otherwise it is a view of self.transformer.
+
+    Examples
+    --------
+    >>> from mvlearn.datasets import load_UCImultifeature
+    >>> from mvlearn.preprocessing import RepeatTransform
+    >>> from sklearn.decomposition import PCA
+    >>> Xs, _ = load_UCImultifeature()
+    >>> repeat = RepeatTransform(PCA(n_components=2))
+    >>> X_transformed = repeat.fit_transform(Xs)
+    >>> print(len(X_transformed))
+    6
+    >>> print(X_transformed[0].shape)
+    (2000, 2)
+    """
+
+    def __init__(self, transformer):
+        self.transformer = transformer
+
+    def _prefit(self, Xs, y=None):
+        r"""Estimate the attributes of the class.
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+             - Xs length: n_views
+             - Xs[i] shape: (n_samples, n_features_i)
+            The data to fit to.
+
+        y : None
+            Ignored variable.
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        Xs = check_Xs(Xs)
+        self.n_views_ = len(Xs)
+        if type(self.transformer) is list:
+            if len(self.transformer) != self.n_views_:
+                raise ValueError(
+                    "The length of the transformers should be the same as the"
+                    "number of views"
+                )
+            self.transformer_list_ = self.transformer
+        else:
+            self.transformer_list_ = [
+                clone(self.transformer) for _ in range(self.n_views_)
+            ]
+        return self
+
+    def fit(self, Xs, y=None):
+        r"""Fit each estimator to the data.
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+             - Xs length: n_views
+             - Xs[i] shape: (n_samples, n_features_i)
+            The data to fit to.
+
+        y : None
+            Ignored variable.
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        self._prefit(Xs, y)
+        for transformer, X in zip(self.transformer_list_, Xs):
+            transformer.fit(X)
+        return self
+
+    def fit_transform(self, Xs, y=None):
+        r"""Fit and transform each dataset
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
```

Comment:
if ndarray all views have the same shape? this is not clear from the docstring

Reply:
Yes they should. It is a problem that we encounter everywhere I guess. I don't think that an error should be returned if a 3d array is passed to `.fit`, which assumes that all views have the same number of features. What do you think @rflperry ?

Reply:
Yes, this the numpy.ndarray correspond to 3d arrays with all views having the same number of features. An error shouldn't be thrown if a 3d array is passed. It's sort of implicit in the second line
```
- Xs length: n_views
- Xs[i] shape: (n_samples, n_features_i)
```
that a 3D ndarray has all the same features. The goal of those two lines was to clarify the shape of multiview data